operators/pkg/instance-controller/controller_test.go

```diff
@@ -290,6 +291,215 @@ var _ = Describe("Instance Operator controller", func() {
+				Memory:                resource.MustParse("1024M"),
+			},
+		}
+		template = crownlabsv1alpha2.Template{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      TemplateName,
+				Namespace: TemplateNamespace,
+			},
+			Spec: crownlabsv1alpha2.TemplateSpec{
+				WorkspaceRef: crownlabsv1alpha2.GenericRef{},
+				PrettyName:   "App Container Template",
+				Description:  "This is the container template",
+				EnvironmentList: []crownlabsv1alpha2.Environment{
+					environment,
+				},
+				DeleteAfter: "30d",
+			},
+			Status: crownlabsv1alpha2.TemplateStatus{},
+		}
+		instance = crownlabsv1alpha2.Instance{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      InstanceName,
+				Namespace: InstanceNamespace,
+			},
+			Spec: crownlabsv1alpha2.InstanceSpec{
+				Template: crownlabsv1alpha2.GenericRef{
+					Name:      TemplateName,
+					Namespace: TemplateNamespace,
+				},
+			},
+			Status: crownlabsv1alpha2.InstanceStatus{},
+		}
+		ns   = v1.Namespace{}
+		tmpl = crownlabsv1alpha2.Template{}
+		inst = crownlabsv1alpha2.Instance{}
+		depl = appsv1.Deployment{}
+	)
+
+	Context("When creating containerized apps", func() {
+		It("Should create the Instance and Template namespaces", func() {
+			ctx = context.Background()
+			By("Creating the Template namespace")
+			Expect(k8sClient.Create(ctx, &templateNs)).Should(Succeed())
+			doesEventuallyExists(ctx, types.NamespacedName{
+				Name: TemplateNamespace,
+			}, &ns, BeTrue(), timeout, interval)
+
+			By("Creating the Instance namespace")
+			Expect(k8sClient.Create(ctx, &instanceNs)).Should(Succeed())
+			Expect(k8sClient.Create(ctx, &webdavSecret)).Should(Succeed())
+			doesEventuallyExists(ctx, types.NamespacedName{
+				Name: InstanceNamespace,
+			}, &ns, BeTrue(), timeout, interval)
+		})
+
+		It("Should create the container Template and the related container Instance", func() {
+			By("Creating the Template")
+			Expect(k8sClient.Create(ctx, &template)).Should(Succeed())
+			doesEventuallyExists(ctx, types.NamespacedName{
+				Name:      TemplateName,
+				Namespace: TemplateNamespace,
+			}, &tmpl, BeTrue(), timeout, interval)
+
+			By("Creating the Instance associated to the Template")
+			Expect(k8sClient.Create(ctx, &instance)).Should(Succeed())
+			doesEventuallyExists(ctx, types.NamespacedName{
+				Name:      InstanceName,
+				Namespace: InstanceNamespace,
+			}, &inst, BeTrue(), timeout, interval)
+		})
+
+		It("Should check the deployment exists", func() {
+			By("Checking for the existence of the deployment")
+			doesEventuallyExists(ctx, types.NamespacedName{
+				Name:      InstanceName,
+				Namespace: InstanceNamespace,
+			}, &depl, BeTrue(), timeout, interval)
+
+			By("Checking the pod list size is not zero")
+			podList := v1.PodList{}
+			Expect(k8sClient.List(ctx, &podList, &client.ListOptions{
+				Namespace: InstanceNamespace,
+			})).Should(Succeed())
+			Expect(podList.Size()).ShouldNot(BeZero())
```

Comment:
This could be avoided, since here you are testing whether Kubernetes works and not your code